src/lib/src/ng-select.component.ts

```diff
@@ -42,10 +43,10 @@ const NGB_ANG_SELECT_VALUE_ACCESSOR = {
         'role': 'dropdown'
     }
 })
-export class NgSelectComponent implements OnInit, OnDestroy, ControlValueAccessor {
+export class NgSelectComponent implements OnInit, OnDestroy, AfterContentInit, ControlValueAccessor {
 
-    @ContentChild(NgOptionDirective) optionTemplateRef: TemplateRef<any>;
-    @ContentChild(NgDisplayDirective) displayTemplateRef: TemplateRef<any>;
+    @ContentChild(NgOptionDirective) optionDirective: NgOptionDirective;
+    @ContentChild(NgDisplayDirective) displayDirective: NgDisplayDirective;
```

Comment:
I think it make sense to rename NgDisplayDirective to NgLabelTemplateDirective and NgOptionTemplateDirective.